### PR TITLE
Move runtime specific NativeRuntimeEventSource code into SPC runtime shim.

### PR DESCRIPTION
**Pull request by lateralusX**
Mono currently fails to AOT NativeRuntimeEventSource after qcalls was added into the shared sources. There are also some Mono specific implementations currently in shared SPC that could be moved.

PR moves all runtime specific implementations of NativeRuntimeEventSource into CoreCLR or Mono specific shim source file part of each runtimes SPC.

Since NativeRuntimeEventSource.ProtableThreadPool.cs only included Mono specific code moved into NativeRuntimeEventSource.ProtableThreadPool.Mono.cs file could be removed from shared part of SPC.

Once Mono starts to support native events we will probably use the same auto-generated methods as CoreCLR and that will reduce the gap minimizing the need for specific shim file implementations in future.

### CODE REVIEW

File: src/mono/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.PortableThreadPool.Mono.cs [hunk 1]
```diff
@@ -13,6 +13,47 @@ namespace System.Diagnostics.Tracing
     // To look at CoreCLR implementation of these events, refer to NativeRuntimeEventSource.PortableThreadPool.CoreCLR.cs.
```
Comment (jkotas) [inline src/mono/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.PortableThreadPool.Mono.cs:13]: Could you keep this file in the shared partition with the CoreCLR condition? It is useful for CoreRT too.

Comment (jkotas) [reply src/mono/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.PortableThreadPool.Mono.cs:13]: (I have done very similar change in CoreRT to fix build breaks - https://github.com/dotnet/runtimelab/pull/707/commits/4089184ac5cee5146b690329d396fa3286124db1 . I just did not have a chance to push it back to dotnet/runtime.)

Comment (lateralusX) [reply src/mono/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.PortableThreadPool.Mono.cs:13]: OK I see.

So if we share what's in NativeRuntimeEventSource.PortableThreadPool.Mono.cs and only include it on none CoreCLR builds you will be fine in CoreRT as well?

If so I will move NativeRuntimeEventSource.PortableThreadPool.Mono.cs in shared and name it to NativeRuntimeEventSource.PortableThreadPool.cs and keep

`<Compile Include="$(MSBuildThisFileDirectory)System\Threading\NativeRuntimeEventSource.PortableThreadPool.cs" Condition="'$(FeatureCoreCLR)' != 'true'" />`

I will add a comment into that file that is currently being shared between Mono and CoreRT to make it more clear and not used on CoreClr.

Comment (jkotas) [reply src/mono/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.PortableThreadPool.Mono.cs:13]: Yes, this works. Thanks!

File: src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs [hunk 2]
```diff
@@ -8,11 +8,52 @@
...
-    // This is part of the NativeRuntimeEventsource for Mono, which is the managed version of the Microsoft-Windows-DotNETRuntime provider.
-    // and contains the implementation of ThreadPool events.
+    // This is part of the NativeRuntimeEventsource for Mono|CoreRT, which is the managed version of the Microsoft-Windows-DotNETRuntime provider.
+    // and contains the implementation of ThreadPool events. Since file is shared between Mono|CoreRT it is kept in shared part of SPC.
     // To look at CoreCLR implementation of these events, refer to NativeRuntimeEventSource.PortableThreadPool.CoreCLR.cs.
     internal sealed partial class NativeRuntimeEventSource : EventSource
     {
+        // On Mono|CoreRT, we don't have these keywords defined from the genRuntimeEventSources.py, so we need to manually define them here.
+        public class Keywords
+        {
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:18]: This file is specific to PortableThreadpool. We should only include the PortableThreadpool-specific keywords here.

(Or undo this change and keep the list under ifdef like it is now.)

Comment (lateralusX) [reply src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:18]: Tried to get away from ifdef:ing these things when moving into runtime specific SPC source files. Do you need these keywords on CoreRT? If not, we could moved them into NativeRuntimeEventSource.PortableThreadPool.Mono.cs shim file closely matching the generated file on CoreClr.

Comment (lateralusX) [reply src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:18]: Looks like most of them are only used in NativeRuntimeEventSource.CoreCLR.cs, except the Threading keyword, so maybe we can exclude all except that one for now.

Comment (jkotas) [reply src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:18]: Yes, that works.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs [hunk 3]
```diff
@@ -8,11 +8,18 @@
...
 {
-    // This is part of the NativeRuntimeEventsource for Mono, which is the managed version of the Microsoft-Windows-DotNETRuntime provider.
-    // and contains the implementation of ThreadPool events.
+    // This is part of the NativeRuntimeEventsource for Mono|CoreRT, which is the managed version of the Microsoft-Windows-DotNETRuntime provider.
+    // and contains the implementation of ThreadPool events. Since file is shared between Mono|CoreRT it is kept in shared part of SPC.
     // To look at CoreCLR implementation of these events, refer to NativeRuntimeEventSource.PortableThreadPool.CoreCLR.cs.
     internal sealed partial class NativeRuntimeEventSource : EventSource
     {
+        // On Mono|CoreRT, we don't have these keywords defined from the genRuntimeEventSources.py, so we need to manually define them here.
+        public class Keywords
```
Comment (marek-safar) [inline src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:17]: nit: should be static

Comment (lateralusX) [reply src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs:17]: Didn't want to deviate from what gets auto generated in runtime\artifacts\obj\coreclr\Eventing\x64\Debug\NativeRuntimeEventSource.CoreCLR.cs:

```
public class Keywords
        {
            public const EventKeywords GCKeyword = (EventKeywords)0x1;
            ....
```

so just followed along in above.
